soc/arm/nxp_imx/rt/Kconfig.soc

```diff
@@ -118,6 +118,37 @@ config SOC_MIMXRT1024
 	select HAS_MCUX_IOMUXC
 	select HAS_SWO
 
+config SOC_MIMXRT1042
+	bool "SOC_MIMXRT1042"
+	select SOC_SERIES_IMX_RT10XX
+	select HAS_MCUX
+	select HAS_MCUX_CACHE
+	select HAS_MCUX_12B1MSPS_SAR
+	select HAS_MCUX_CCM
+	select HAS_MCUX_ENET
+	select HAS_MCUX_FLEXSPI
+	select HAS_MCUX_IGPIO
+	select HAS_MCUX_LPI2C
+	select HAS_MCUX_LPSPI
+	select HAS_MCUX_LPUART
+	select HAS_MCUX_GPT
+	select HAS_MCUX_SEMC
+	select HAS_MCUX_TRNG
```

Comment:
Can we delete all of these configs?

Reply:
I believe I have removed the Kconfigs that can be stripped here. Some still seem to be required. I checked this by using `grep` to search for the Kconfig symbol, and if it appeared outside of the `soc` directory (or in C source files within the SOC) directory, I assumed the Kconfig was still required. We should likely do a larger cleanup PR to remove these Kconfigs though, so that these are not introduced with additional SOCs.